ui/src/components/song/SongContainer.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  Box,
+  Button,
+  Container,
+  Stack,
+  Typography,
+  TextField,
+  IconButton,
+  Modal,
+} from '@mui/material';
+import axios from 'axios';
+import { FC, ReactElement } from 'react';
+import { useEffect, useState, useCallback } from 'react';
+import { SongCardProps, SongSearchFilter } from '../../types/song';
+import SongCard from './SongCard';
+import SongSearch from './SongSearch';
+
+const SongContainer: FC = (): ReactElement => {
+  const [allSongs, setAllSongs] = useState([] as SongCardProps[]);
```

Comment:
this is the proper way: `useState<SongCardProps[]>([])`

please apply this to all the state definitions you have